docs: rework Browser chapter (#503)
oasisprotocol/sapphire-paratime: docs/develop/browser/README.md

[ethers-api]: https://api.docs.oasis.io/js/sapphire-ethers-v6
[ethers-github]: https://github.com/oasisprotocol/sapphire-paratime/tree/main/integrations/ethers-v6
[viem-npmjs]: https://www.npmjs.com/package/@oasisprotocol/sapphire-viem-v2
[viem-api]: https://api.docs.oasis.io/js/sapphire-viem-v2
[viem-github]: https://github.com/oasisprotocol/sapphire-paratime/tree/main/integrations/viem-v2
[wagmi-npmjs]: https://www.npmjs.com/package/@oasisprotocol/sapphire-wagmi-v2
[wagmi-api]: https://api.docs.oasis.io/js/sapphire-wagmi-v2
[wagmi-github]: https://github.com/oasisprotocol/sapphire-paratime/tree/main/integrations/wagmi-v2

## Choosing the Right Library

Many browser-based dApps can use the lightweight
[Sapphire TypeScript wrapper][s-p] if they rely entirely on the
injected EIP-1193 wallet provider (e.g. window.ethereum) to communicate with
and sign transactions on Sapphire. If you already use an EVM-frontend library,
use our library-specific packages such as [Ethers][ethers], [Viem][viem] or
[Wagmi][wagmi].

[s-p]: ./README.md#lightweight-sapphire-typescript-wrapper
[ethers]: ./ethers.md
[viem]: ./viem.md
[wagmi]: ./wagmi.md

:::info Example: Starter project

> If your project includes both a smart contract backend and a web frontend, you
> can explore our [demo-starter] repository. It provides a working example using
> React and also includes a Vue branch.

Reviewer: ```suggestion
If your project includes both a smart contract backend and a web frontend, you
can explore our [demo-starter] repository. It provides a working example using
React as well as a Vue branch.
```

Maybe we should also link https://github.com/oasisprotocol/demo-starter/tree/vue

:::

[demo-starter]: https://github.com/oasisprotocol/demo-starter

## Transaction encryption

> When using the wrapper libraries, ensure that all transactions containing

Reviewer: ```suggestion
When using the supported libraries, ensure that all transactions containing
```

I think of supported libraries as the total set, and wrapper as just the JS client? Very pedantic, but I was wondering since we introduced the term "Supported Libraries" earlier.

sensitive information **are encrypted**. Encryption is essential to safeguard user
data and ensure privacy. To verify that a transaction is encrypted, you can
check the transaction details on the Oasis Block Explorer for the corresponding
network ([Localnet], [Testnet], or [Mainnet]). Look for a green lock icon next
to the transaction, which indicates that it is securely encrypted.

:::tip View-Call Authentication

For authenticated view calls, make sure to visit the [View-Call Authentication]
chapter to learn about the proper authentication procedures.

:::

[Localnet]: http://localhost:8548
[Testnet]: https://explorer.oasis.io/testnet/sapphire
[Mainnet]: https://explorer.oasis.io/mainnet/sapphire
[View-Call Authentication]: ../authentication.md

## Lightweight Sapphire TypeScript Wrapper

> The Sapphire TypeScript wrapper from our
> [`@oasisprotocol/sapphire-paratime`][s-p-github] library will automatically
> encrypt the eth_call, eth_estimateGas and eth_signTransaction JSON-RPC calls.

Reviewer: ```suggestion
The Sapphire TypeScript wrapper from our
[`@oasisprotocol/sapphire-paratime`][s-p-github] library will automatically
encrypt the `eth_call`, `eth_estimateGas` and `eth_signTransaction` JSON-RPC calls.
```

### Usage

> Install the library via your favorite package manager

Reviewer: ```suggestion
Install the library with your favorite package manager
```

```shell npm2yarn
npm install -D @oasisprotocol/sapphire-paratime
```

After installing this library, find your Ethereum provider and wrap it using
wrapEthereumProvider.

```js
import { wrapEthereumProvider } from '@oasisprotocol/sapphire-paratime';

const provider = wrapEthereumProvider(window.ethereum);
```

:::info Example: Hardhat boilerplate

> Our maintained Hardhat boilerplate uses the Sapphire TypeScript wrapper to

Reviewer: I know we talked about this [here](https://github.com/oasisprotocol/sapphire-paratime/issues/283#issuecomment-2337694120), but I still want to remove the Hardhat boilerplate.

Author: noted, though I don't think we need to solve it in this PR.

[ethers]: ./ethers.md
[viem]: ./viem.md
[wagmi]: ./wagmi.md

:::info Example: Starter project

If your project involves building both a contract backend and a web frontend,
we recommend checking out our [demo-starter] repository.

[demo-starter]: https://github.com/oasisprotocol/demo-starter

:::

## Transaction encryption

When using the wrapper libraries, ensure that all transactions containing
sensitive information are encrypted. Encryption is essential to safeguard user
data and ensure privacy. To verify that a transaction is encrypted, you can
check the transaction details on the Oasis Block Explorer for the corresponding
network ([localnet], [testnet], or [mainnet]). Look for a green lock icon next
to the transaction, which indicates that it is securely encrypted.

:::tip View-Call Authentication

For authenticated view calls, make sure to visit the [View-Call Authentication]
chapter to learn about the proper authentication procedures.

:::

[localnet]: http://localhost:8548
[testnet]: https://explorer.oasis.io/testnet/sapphire
[mainnet]: https://explorer.oasis.io/mainnet/sapphire
[View-Call Authentication]: ../authentication.md

> ## Lightweight Sapphire TypeScript Wrapper

Author: I think having a quick note of the TypeScript wrapper is good, so that without a media break (switching to GitHub), the developer gets a feel for how to use the simple integration.
I shortened the section to focus just on that.
